pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -337,12 +337,21 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
             publishContext.setOriginalHighestSequenceId(highestSequenceId);
             headersAndPayload.readerIndex(readerIndex);
         }
-
+        long chunkID = 0;
+        if (publishContext.isChunked()) {
+            headersAndPayload.markReaderIndex();
+            MessageMetadata msgMetadata = Commands.parseMessageMetadata(headersAndPayload);
```

Comment:
You are parsing the metadata twice here.

Define `MessageMetadata md = null;` outside at line 323.

and do parse here again if md is null.